processor/routingprocessor/traces.go

```diff
@@ -105,7 +106,7 @@ func (p *tracesProcessor) route(ctx context.Context, t ptrace.Traces) error {
 		matchCount := len(p.router.routes)
 		for key, route := range p.router.routes {
 			_, isMatch, err := route.statement.Execute(ctx, stx)
-			if err != nil {
+			if err != nil && p.config.ErrorMode == ottl.PropagateError {
```

Comment:
@kovrus note, for each signal, if the error happens when executing the statement's function, `isMatch` will be true and `err` will be non-nil.  If using `ignore`, `p.group(key, groups, route.exporters, rspans)` will be called, but whatever the goal with the statement's function was will not have happened, since the function errored.  Is that acceptable?

Reply:
I assumed that isMatch is `false` when the error happens. I think we need to adjust the routing logic then. So now we send resource metrics to the default exporter when there is no match across all route definitions or when there is at least one failure under the condition that error mode is `ignore`. Something like that should do the job, we can refine this logic later: 

 
```go
		matchCount := len(p.router.routes)
		for key, route := range p.router.routes {
			_, isMatch, err := route.statement.Execute(ctx, stx)
			if err != nil && p.config.ErrorMode == ottl.PropagateError {
				return err
			}

			if err != nil && p.config.ErrorMode == ottl.IgnoreError {
				p.group("", groups, p.router.defaultExporters, rspans)
				continue
			}

			if !isMatch {
				matchCount--
				continue
			}
			p.group(key, groups, route.exporters, rspans)
		}

		if matchCount == 0 {
			// no route conditions are matched, add resource spans to default exporters group
			p.group("", groups, p.router.defaultExporters, rspans)
		}
```

cc: @jpkrohling

Reply:
`IsMatch` always reflects the output of the condition; if the condition errored then it is false, but if the condition passed it will be true, even if the Invocation errors.

I will update the functions to match your desired logic.

Reply:
Updated